src/cache.js

```diff
@@ -0,0 +1,67 @@
+const Apigee = require('./lib/apigee')
+const yaml = require('js-yaml')
+const fs = require('fs')
+const chai = require('chai')
+const expect = chai.expect
+
+const isUpdated = (a, b, properties) => {
+  return properties.find(prop => {
+    if (prop.includes('.')) {
+      const props = prop.split('.')
+      return a[props[0]][props[1]] !== b[props[0]][props[1]]
+    }
+    return a[prop] !== b[prop]
+  })
+}
+
+const validateDate = (cache) => {
+  if ((cache.timeoutInSec !== null && cache.timeOfDay !== null) ||
+     (cache.timeOfDay !== null && cache.expiryDate !== null) ||
+     (cache.timeoutInSec !== null && cache.expiryDate !== null)) {
+      console.log('You can only have 1 expirySettings option. Set to default 300 seconds')
+    return {'timeoutInSec': {'value': 300}, 'valuesNull': cache.expirySettings.valuesNull}
+  } else if (cache.expiryDate) {
+      return {'expiryDate': {'value': cache.expiryDate}, 'valuesNull': cache.expirySettings.valuesNull}
+  } else if (cache.timeOfDay) {
+      return {'timeOfDay': {'value': cache.timeOfDay}, 'valuesNull': cache.expirySettings.valuesNull}
+  } else if (cache.timeoutInSec) {
+      return {'timeoutInSec': {'value': cache.timeoutInSec}, 'valuesNull': cache.expirySettings.valuesNull}
+  } else {
+      return {'timeoutInSec': {'value': 300}, 'valuesNull': cache.valuesNull}
+  }
+}
+
+module.exports = async (config, manifest) => {
+  const apigee = new Apigee(config)
+  let yml = yaml.safeLoad(fs.readFileSync(manifest, 'utf8'))
+  const cacheConfig = yml.caches
+  console.log(yml.caches)
+  if (!cacheConfig) {
+    return false
+  }
+  Object.keys(cacheConfig).map(async (cacheName) => {
```

Comment:
cacheName hier is geen name, maar gewoon een volgnummer, aangezien cacheConfig al een array is. Je kunt hier dus gewoon loopen over cacheConfig:
`cacheConfig.map(async(cache) => {`

Reply:
En als er een object in die array zit( als enige value)? Dit zag ik namelijk precies hetzelfde bij de kvms.js file gebeuren (daarom dat ik het zo toegepast heb)